src/platform/forms-system/src/js/utilities/data/reduceErrors.js

```diff
@@ -0,0 +1,156 @@
+ *
+ * min/max: {
+    property: 'instance.someProperty',
+    message: 'does not meet minimum length of 1',
+    name: 'minItems',
+    argument: 1,
+    schema: {...},
+    stack: 'instance.someProperty does not meet minimum length of 1'
+  }
+  * required: {
+    property: 'instance',
+    message: 'requires property "someProperty"',
+    name: 'required',
+    argument: 'someProperty',
+    schema: {...},
+    stack: 'instance requires property "someProperty"'
+  }
+  * array: {
+    property: 'instance.newDisabilities[0]',
+    message: 'requires property "cause"',
+    name: 'required',
+    argument: 'cause',
+    schema: {...},
+    stack: 'instance.newDisabilities[0] requires property "cause"'
+  }
+  * non-empty "__errors" array: {
+    __errors: ['error message']
+  }
+*/
+/**
+ * @typedef Form~formErrors
+ * @type {object}
+ * @property {Form~rawErrors}
+ * @property {Form~errors}
+ */
+/**
+ * Process rawErrors from jsonschema validator into a more friendly list
+ * @param {Form~formErrors} errors
+ * @param {object[]} pageList - list of all form pages from `form.pages`
+ */
+export const reduceErrors = (errors, pageList) =>
+  errors.reduce((acc, error) => {
+    const findErrors = (key, err) => {
+      if (typeof err === 'object') {
+        if (err.message) {
+          const property = err.property.replace(/instance\.?/, '') || '';
+          const instance = property.replace(/(\[\d+\])/, '');
+          const name = err.name === 'required' ? err.argument : instance;
+          /*
+           * There may be multiple errors for the same property, e.g.
+           * `contestedIssues` in form 996. It can have a custom message
+           * "Please select a contested issue" and a minItem error
+           * "Does not meet minimum length of 1"; there's no need to confuse
+           * anyone and show both
+          */
+          if (!errorExists(acc, name)) {
+            const { chapterKey = '', pageKey = '' } = getPropertyInfo(
+              pageList,
+              err.argument,
+              err.property.startsWith('instance.') ? instance : '',
+            );
+            acc.push({
+              name,
+              // property may be `array[0]`; we need to extract out the `[0]`
+              index: property.match(/\[(\d+)\]/)?.[1] || null,
+              message: err.stack,
+              chapterKey,
+              pageKey,
+            });
+          }
+          return null;
+        }
+        if (err.__errors && err.__errors.length && !errorExists(acc, key)) {
+          const { chapterKey = '', pageKey = '' } = getPropertyInfo(
+            pageList,
+            key,
+          );
+          acc.push({
+            name: key,
+            index: null,
+            message: err.__errors.join('. '),
+            chapterKey,
+            pageKey,
+          });
+        }
+        Object.keys(err).forEach(sub => findErrors(sub, err[sub]));
+      }
+      return null;
+    };
+    findErrors('', error);
```

Comment:
Why first call this recursive function with `key = ''`?

Reply:
Honestly, I don't remember why I initialized it with an empty string. I'll make it a more appropriate call.